### ISSUE TITLE
VirtualService sslConfig pointing to a SDS not generating correct envoy config

### ISSUE THREAD

**Issue by pszeto** [solo-io/gloo#6799]
### Gloo Edge Version

1.11.x (latest stable)

### Kubernetes Version

1.22.x

### Describe the bug

Configuring a Virtual Service to use an SDS to get certificates not generating correct envoy config.

### Steps to reproduce the bug

1. Create the following VS.

```
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: vs-direct
  namespace: gloo-system
spec:
  sslConfig:
    sds:
      targetUri: unix:/tmp/uds_path
      certificatesSecretName: upstream-tls
  virtualHost:
    domains:
      - '*'
    routes:
      - matchers:
         - prefix: /
        directResponseAction:
          status: 200
          body: "Hello, world!"
```

The envoy config has:

```
transport_socket:
                  name: envoy.transport_sockets.tls
                  typed_config:
                    '@type': >-
                      type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext
                    common_tls_context:
                      tls_params: {}
                      alpn_protocols:
                        - h2
                        - http/1.1
                      tls_certificate_sds_secret_configs:
                        - name: upstream-tls
                          sds_config:
                            api_config_source:
                              api_type: GRPC
                              grpc_services:
                                - envoy_grpc:
                                    cluster_name: gateway_proxy_sds
                              transport_api_version: V3
                            resource_api_version: V3
```

### Expected Behavior

Expected the envoy config to look similiar :

https://www.envoyproxy.io/docs/envoy/latest/configuration/security/secret.html

```
      typed_config:
        "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
        common_tls_context:
          tls_certificate_sds_secret_configs:
          - name: client_cert
            sds_config:
              resource_api_version: V3
              api_config_source:
                api_type: GRPC
                transport_api_version: V3
                grpc_services:
                  google_grpc:
                    target_uri: unix:/tmp/uds_path
```

**Comment by nfuden** [comment]
See linked zendesk for more info, we this is more about secret retrieval from vault which then leads to this

**Comment by kdorosh** [comment]
another option @pszeto to unblock on the current version is to define an upstream with the pipe location to the SDS server, and then configure SDS to use the generated envoy cluster (with that pipe)

**Comment by pszeto** [comment]
@kdorosh 
Yes, their aim is to load downstream tls certs from Vault.

**Comment by pszeto** [comment]
@kdorosh Tried the work around:
```
apiVersion: gloo.solo.io/v1
kind: Upstream
metadata:
  name: sds-sandbox
  namespace: gloo-system
spec:
  pipe:
    path: /tmp/solo-sds.sock
    serviceSpec:
      grpc: {}
```

and changed a VirtualService to have
```
spec:
  sslConfig:
    sds:
      certificatesSecretName: temp.solo.com
      clusterName: sds-sandbox_gloo-system
    sniDomains:
    - temp.solo.com
  ```
  
  Got the following error:
  ```
[2022-08-09 18:30:21.550][7][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:126] gRPC config for [type.googleapis.com/envoy.config.listener.v3.Listener](http://type.googleapis.com/envoy.config.listener.v3.Listener) rejected: Error adding/updating listener(s) listener-::-8443: envoy.config.core.v3.ApiConfigSource must have a statically defined non-EDS cluster: 'sds-sandbox_gloo-system' does not exist, was added via api, or is an EDS cluster
```

**Comment by kdorosh** [comment]
@pszeto we can do this by adding the cluster to the envoy bootstrap config (the configmap) rather than using an upstream

**Comment by bewebi** [comment]
A fix for this bug, which ensures that an appropriate `google_grpc` `grpc_service` is created when `targetUri` are specified in `sslConfig`, has been merged to master and all supported LTS branches.

The fix is included in the new v1.11.29 release and will be a part of the next release for minor versions 1.9, 1.10, 1.12, and 1.13-beta

**Comment by bewebi** [comment]
The fix has now been released in Gloo EE v1.11.34
